### PR TITLE
Addresses #3 Testing using MockBackend for KnowledgeAPI service

### PR DESCRIPTION
**Pull request by harshit98**
Fixes issue #3 

Changes: 
- Added tests for knowledgeApi service so that no more issues related to API occurs.
- Removed unnecessary imports of the modules which were causing errors.
- Code indentation.
- Created a `shared/mock-backend/knowledge.mock.ts`. We can further add more mocks in the mock-backend folder.

Demo Link: (Not appropriate here)

Problems I'm facing: 
While testing, it throws error `Expected 'http://lookup.dbpedia.org/api/search/KeywordSearch?&QueryString=Berlin' to be 'http://lookup.dbpedia.org/api/search/KeywordSearch?QueryString=Berlin'.`

UPDATE : Solved

Kindly review: @nikhilrayaprolu @Marauderer97 If you have any solution for my problem, let me know. : )

### CODE REVIEW

File: src/app/about/about.component.spec.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* tslint:disable:no-unused-variable */
```
Comment (nikhilrayaprolu) [inline src/app/about/about.component.spec.ts]: what made you add this here

File: src/app/auto-complete/auto-complete.component.spec.ts [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
-import {FormsModule} from "@angular/forms";
-import {CommonModule} from "@angular/common";
-import {BrowserModule} from "@angular/platform-browser";
-import {RouterTestingModule} from "@angular/router/testing";
-import { ThemeComponent } from '../theme/theme.component';
-import { DropdownComponent } from '../dropdown/dropdown.component';
+import { AutocompleteService } from '../autocomplete.service';
+
+/**
+ * To avoid error 'No provider for Http'
```
Comment (nikhilrayaprolu) [inline src/app/auto-complete/auto-complete.component.spec.ts]: I don't think we need this comments.

File: src/app/knowledgeapi.service.spec.ts [hunk 3]
```diff
@@ -1,25 +1,75 @@
...
+  it('should call knowledge service API and return the result', () => {
+    backend.connections.subscribe((connection: MockConnection) => {
+      const options = new ResponseOptions({
+        body: JSON.stringify(MockKnowledgeApi)
+      });
+    
+      connection.mockRespond(new Response(options));
+      expect(connection.request.method).toEqual(RequestMethod.Get);
+      expect(connection.request.url).toBe(
+        `http://lookup.dbpedia.org/api/search/KeywordSearch` +
```
Comment (nikhilrayaprolu) [inline src/app/knowledgeapi.service.spec.ts]: the error you have raised is being called from here, so might be you '&' before QueryString

**Comment by harshit98** [comment]
@nikhilrayaprolu kindly review. : )

- I have removed comments.
- Removed `tslint:disable`
- Resolved the error which was occurring.

**Comment by nikhilrayaprolu** [comment]
@Marauderer97 can you review this, and suggest around any more tests?

**Comment by harshit98** [comment]
> Check if title corresponds with query

Already done. This test fails when DBpedia does not respond. There is line in code `?&QueryString=${searchquery}` which will verify it.

Other points which you have highlighted, I'll solve them in separate PR. They will be added in infobox component. @nikhilrayaprolu it would be great if you can provide your inputs regarding this discussion.

**Comment by nikhilrayaprolu** [comment]
@harshit98 looks good.since this is related to knowledge service.let us make another pull for infobox component.

**Comment by nikhilrayaprolu** [comment]
@harshit98 please resolve conflicts.

**Pull request merged**